sotodlib/io/load_book.py

```diff
@@ -0,0 +1,223 @@
+import so3g
+from spt3g import core as spt3g_core
+import yaml
+import os
+
+import sotodlib
+from sotodlib import core
+
+#from . import load
+
+import numpy as np
+
+def _extract_1d(src, src_offset, dest, dest_offset):
```

Comment:
Ya, you caught me on this ... I will see if there are more things I could take advantage of.

The existing io.load stuff was written in the G3TimestreamMap days and contains many intrinsic inefficiencies because of that.  But I should be more careful about that.

Reply:
Ahh gotcha. I'm not against updating what's in `io.load` to be more efficient and having that flow down into updates for g3tsmurf loading, etc. I think we're all using SuperTimestreams at this point.

Mostly interested in keeping the same functions/structure.

Reply:
The major difference is that this loader assumes it knows exactly how many samples it is supposed to load, and how many detectors.  That means it needs a well-constructed obsfiledb... but with those in hand it is very easy to preallocate the data, and not have to use hstack anywhere.